src/components/User/UserTrainingModels.tsx

```diff
@@ -109,20 +116,53 @@ const trainingStatusFields: Record<TrainingStatus, { color: MantineColor; descri
   },
 };
 
+const modelsLimit = 10;
+
 const minsWait = 5 * 60 * 1000;
 const minsPerEpoch = 1 * 60 * 1000;
 const minsPerEpochSDXL = 5 * 60 * 1000;
 
+const TrainingSignals = () => {
+  const queryClient = useQueryClient();
+
+  const onUpdate = useCallback(
+    (updated: TrainingUpdateSignalSchema) => {
+      const queryKey = getQueryKey(trpc.model.getMyTrainingModels);
+      queryClient.setQueriesData(
+        { queryKey, exact: false },
+        // TODO [bw] can we type "old" better here?
+        produce((old: any) => {
```

Comment:
Not sure if it'll actually work, but you can export the response type from the `getMyTrainingModels` endpoint inside `~/types/router`
```ts
export type ModelGetMyTrainingModels = ModelRouter['getMyTrainingModels']['items'];
```
Then here we can type it
```suggestion
        produce((old: ModelGetMyTrainingModels | undefined) => {
```
Give it a try

Reply:
yep that works, we should do that in more places (usually we just have old, untyped)

Reply:
I think in this case is marked as any because you're using the query client directly instead of the trpc context, so it doesn't know which endpoint to match